### ISSUE TITLE
IDE0039 false-positive in expression tree

### ISSUE THREAD

**Issue by yaakov-h** [dotnet/roslyn#23708]
**Version Used**: 
Microsoft Visual Studio 2017 Enterprise v15.5

**Steps to Reproduce**:

1.  Load the following code into the editor:

```cs
using System;
using System.Collections.Generic;
using System.Linq.Expressions;

namespace RoslynSandbox
{
	public class Class1
	{
		public static void Foo()
		{
			Predicate<object> predicate = _ => true;
			GetFiltered<object>(o => predicate(o));
		}

		static IEnumerable<T> GetFiltered<T>(Expression<Predicate<T>> expression) => default;
	}
}
```

**Expected Behavior**:

The above code compiles with no error, warnings, or suggestions.

**Actual Behavior**:

Suggestion IDE0039 "Use local function" on line 11.

If I apply the suggested code fix, the code becomes:

```cs
using System;
using System.Collections.Generic;
using System.Linq.Expressions;

namespace RoslynSandbox
{
	public class Class1
	{
		public static void Foo()
		{
			bool predicate(object _) => true;
			GetFiltered<object>(o => predicate(o));
		}

		static IEnumerable<T> GetFiltered<T>(Expression<Predicate<T>> expression) => default;
	}
}
```

This code does **not** compile, due to the following error:

```
error CS8110: An expression tree may not contain a reference to a local function
```

**Comment by CyrusNajmabadi** [comment]
Fixed with https://github.com/dotnet/roslyn/pull/23136

**Comment by sharwell** [comment]
Duplicate of #22672

**Comment by yaakov-h** [comment]
Confirmed resolved with Visual Studio 2017 Enterprise v15.6 Preview 1.0
